version: '3.9'
services:
  readme.user.postgres:
    image: postgres:17-alpine
    container_name: readme.user.postgres
    environment:
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
      POSTGRES_DB: ${POSTGRES_DB}
    ports:
      - 5434:5432
    volumes:
      - readme.postgres.user:/var/lib/postgresql/data

    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U ${POSTGRES_USER} -d {POSTGRES_DB}" ]
      interval: 10s
      timeout: 5s
      retries: 5
      start_period: 10s

    networks:
      - readme.network

  readme.user.pgadmin:
    image: dpage/pgadmin4
    container_name: readme.user.pgadmin
    environment:
      PGADMIN_DEFAULT_EMAIL: ${PGADMIN_DEFAULT_EMAIL}
      PGADMIN_DEFAULT_PASSWORD: ${PGADMIN_DEFAULT_PASSWORD}
      PGADMIN_CONFIG_SERVER_MODE: "False"
    ports:
      - 8082:80
    depends_on:
      - readme.user.postgres
    networks:
      - readme.network

volumes:
  readme.postgres.user:

networks:
  readme.network:
      driver: bridge
